File: packages/core/tests/fixtures/unhealthy/complex.php
<?php
function processData($user, $config, $logger, $db, $cache): array {
    $result = [];
    if ($user->isActive()) {
        foreach ($db->getItems() as $item) {
            switch ($item->status) {
                case 'active':
                    if ($item->priority > 10) {
                        while ($cache->isLocked()) {
                            sleep(1);
                        }
                        $logger->log($item->id);
                        $result[] = $item;
                    }
                    break;
                case 'pending':
                    if ($item->createdAt < $config->cutoff) {
                        throw new Exception('Too old');
                    }
                    break;
            }
        }
    }
    return $result;
}

File: packages/core/tests/fixtures/unhealthy/complex.rb
def process_data(user, config, logger, db, cache)
  result = []
  if user.active?
    db.get_items.each do |item|
      case item.status
      when :active
        if item.priority > 10
          while cache.locked?
            sleep(0.1)
          end
          logger.log(item.id)
          result << item
        end
      when :pending
        if item.created_at < config.cutoff
          raise "Too old"
        end
      end
    end
  end
  result
end
